ROSNET/ROSMessageParser/MessageDefinitionParser.cs

```diff
@@ -0,0 +1,263 @@
+                fieldValuesBySubDefinitionName.Add(name, fields);
+
+                //Adds copy of subdefinition to the dictionary with last name of the subdefinition (dictionary contains std_msgs/Header and Header). Some definitions only use last name.
+                if (name.Contains("/"))
+                {
+                    fieldValuesBySubDefinitionName.Add(name.Split("/").Last(), fields);
+                }
+            }
+
+            return ParseMainDefinition(mainDefinition,fieldValuesBySubDefinitionName);
+
+
+        }
+
+        /// <summary>
+        /// Parses a subdefinition in the messagedefinition
+        /// </summary>
+        /// <returns>name of subdefinition and list of fields in subdefinition</returns>
+        private static (string, List<FieldValue>) ParseSubDefinition(string subDefinition, Dictionary<string, List<FieldValue>> fieldValuesByDefinitionName )
+        {
+            //gets the definitionname from first line and skips "MSG: " before the name
+            string definitionName = subDefinition.Split("\n").First().Split(" ").Last();
+            var lines = subDefinition.Split("\n").Skip(1);
+
+            List<FieldValue> fieldValues = ParseDefinition(lines, fieldValuesByDefinitionName);
+
+            return (definitionName, fieldValues);
+        }
+
+        /// <summary>
+        /// Parses the maindefinition using subdefinitions
+        /// </summary>
+        /// <returns>list of fields in mainmessage</returns>
+        private static List<FieldValue> ParseMainDefinition(string mainDefinition, Dictionary<string, List<FieldValue>> fieldValuesByDefinitionName)
+        {
+            string[] lines = mainDefinition.Split("\n");
+
+            List<FieldValue> fieldValues = ParseDefinition(lines, fieldValuesByDefinitionName);
+
+            return fieldValues;
+
+
+        }
+
+        /// <summary>
+        /// Parses a definition using subdefinitions
+        /// </summary>
+        /// <returns>list of fields in definition</returns>
+        private static List<FieldValue> ParseDefinition(IEnumerable<string> lines, Dictionary<string, List<FieldValue>> fieldValuesByDefinitionName)
+        {
+            var validLines = new List<String>();
+            var commentRegex = new Regex(@"#.*");
+            foreach (string line in lines)
+            {
+                string tempLine = commentRegex.Replace(line, "").Trim(); //removes comments
+                if (!string.IsNullOrWhiteSpace(tempLine)) //removes empty lines
+                {
+                    validLines.Add(tempLine);
+                }
+            }
+
+            var definitionFields = new List<FieldValue>();
+            var arrayRegex = new Regex(@".*\[\]");
+            var fixedLengthArrayRegex = new Regex(@".*\[[0-9]+\]");
+            foreach (var line in validLines)
+            {
+                var wordsInLine = line.Split(" ").SelectMany(t => t.Split("=")); //splits field into words
```

Comment:
Can this be factored out into a "parseLine"-method?

Reply:
Men som jeg nevnte på Slack, så burde alt dette egt. bare byttes ut med deler fra https://github.com/siemens/ros-sharp